leetcode/128_longest_consecutive_sequence.py

```diff
@@ -0,0 +1,27 @@
+from typing import List
+
+class Solution:
+    def longestConsecutive(self, nums: List[int]) -> int:
+        """Finds the length of the longest consecutive sequence in an unsorted list of integers.
+
+        Args:
+            nums (List[int]): A list of integers.
+
+        Returns:
+            int: The length of the longest consecutive sequence of integers.
+        """
+
+        num_set = set(nums)
+        longest = 0
+
```

Comment:
```
[9,0,0,0,0,0,0,0,0,0,0,0,0,0,3,7,2,5,8,4,6,0,1]
```

skip everything except 0
0 check 1-n/2
0 check 1-n/2
n/2 number of 0s
 
Solutions
* Store the max sequence length in a separate set/map. Then skip if we've already seen it.
* Convert nums to a set at the start. Then iterate over `num_set`

longest[n] = longest[n+1] + 1
longest[4] = longest[5] + 1
longest[5] = 3